### DISCUSSION TITLE
crazyflies take off and fall immediately! (seems like there's problem with the pitch angle)

### DISCUSSION THREAD

**Discussion by Xuxxuuu** [bitcraze/discussions#1270] [Q&A]
l deployed the crazychoir package(a ROS 2 toolbox allowing users to run simulations and experiments on swarms of Crazyflie [nano-quadrotors).](https://opt4smart.github.io/crazychoir/) ,this package built the crazyflie firmware and created python bindings already.
Now I run the tracking_vicon.launch.py and start takeoff. Crazyflie takes off and falls immediately while it seems like turn over(which i think there's something wrong with the pitch angle). I use a video link to show this phenomenon. I am certain that the crazyflie has sufficient battery and the propellers are intact.
Here is also a screenshot of running the launch file and there's no error prompt.
![roslaunch](https://github.com/bitcraze/discussions/assets/161097297/c808a959-6e3d-4276-8ffc-dae7315d82d8)
I wonder whether the problem is about the crazyflie firmware or the crazychoir package. l have adjusted the parameters in crazychoir package but the crazyflie falls in the same way.
Here is the code about sending thrust to the [crazyflie：](https://github.com/OPT4SMART/crazychoir/tree/master/crazychoir/crazychoir/radio_handler)
It would be great if you could provide some guidance!
[](https://www.youtube.com/shorts/Spq0xaM-fcg)
[](https://www.youtube.com/shorts/2gI3BO_IuMc)

**Comment by Xuxxuuu** [comment]
> l deployed the crazychoir package(a ROS 2 toolbox allowing users to run simulations and experiments on swarms of Crazyflie [nano-quadrotors).](https://opt4smart.github.io/crazychoir/) ,this package built the crazyflie firmware and created python bindings already. Now I run the tracking_vicon.launch.py and start takeoff. Crazyflie takes off and falls immediately while it seems like turn over(which i think there's something wrong with the pitch angle). I use a video link to show this phenomenon. I am certain that the crazyflie has sufficient battery and the propellers are intact. Here is also a screenshot of running the launch file and there's no error prompt. ![roslaunch](https://private-user-images.githubusercontent.com/161097297/320249727-c808a959-6e3d-4276-8ffc-dae7315d82d8.jpg?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTI0NjAwMzYsIm5iZiI6MTcxMjQ1OTczNiwicGF0aCI6Ii8xNjEwOTcyOTcvMzIwMjQ5NzI3LWM4MDhhOTU5LTZlM2QtNDI3Ni04ZmZjLWRhZTczMTVkODJkOC5qcGc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNDA3JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDQwN1QwMzE1MzZaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0xYjI3ZjVmNTI0ZTA5ZmJiNjJjODY1NmFjOTIzNDg4YTVjMDM4MjQ5ZTVjNmUzZGU3ZDFmZWYxYmEwZTMwYTQxJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.umkK5NvzX_5psQNcPOOQ2KqJom3JEZrpbCpLS1qSSAY) I wonder whether the problem is about the crazyflie firmware or the crazychoir package. l have adjusted the parameters in crazychoir package but the crazyflie falls in the same way. Here is the code about sending thrust to the [crazyflie：](https://github.com/OPT4SMART/crazychoir/tree/master/crazychoir/crazychoir/radio_handler) It would be great if you could provide some guidance!

the video：https://www.youtube.com/shorts/2gI3BO_IuMc
https://www.youtube.com/shorts/Spq0xaM-fcg

**Comment by gemenerik** [comment]
This appears to be a problem with the CrazyChoir ROS2 toolbox which is not developed by Bitcraze, please open an issue on the CrazyChoir GitHub page.
